src/Imath/half.h

```diff
@@ -15,7 +15,394 @@
+///
+/// Testing on a Core i9, the timings are approximately:
+///
+/// half to float table: 0.69 ns / call
+/// small table: 0.93 ns / call
+/// no table: 1.36 ns / call
+/// f16c: 0.45 ns / call
+///
+/// original: 5.2 ns / call
+/// float to half: 4.57 ns / call**
+/// no table: 1.48 ns / call**
+/// f16c: 0.45 ns / call
+///
+/// ** - this depends on the distribution of the floats in question
+///
+/// Further, an implementation wishing to receive floating point
+/// exceptions on underflow / overflow when converting float to half
+/// can include this file with IMATH_HALF_EXCEPTIONS_ENABLED defined.
+///
+#if defined(__has_include)
+#    if __has_include(<x86intrin.h>)
+#        include <x86intrin.h>
+#    elif __has_include(<intrin.h>)
+#        include <intrin.h>
+#    endif
+#endif
+
+//-------------------------------------------------------------------------
+// Limits
+//
+// Visual C++ will complain if HALF_DENORM_MIN, HALF_NRM_MIN etc. are not float
+// constants, but at least one other compiler (gcc 2.96) produces incorrect
+// results if they are.
+//-------------------------------------------------------------------------
+
+#if (defined _WIN32 || defined _WIN64) && defined _MSC_VER
+
+/// Smallest positive denormalized half
+#    define HALF_DENORM_MIN 5.96046448e-08f
+
+#    define HALF_NRM_MIN 6.10351562e-05f // Smallest positive normalized half
+
+#    define HALF_MIN 6.10351562e-05f // Smallest positive normalized half
+
+#    define HALF_MAX 65504.0f // Largest positive half
+
+#    define HALF_EPSILON                                                                           \
+        0.00097656f // Smallest positive e for which                                               \
+                    // half (1.0 + e) != half (1.0)
+#else
+
+#    define HALF_DENORM_MIN 5.96046448e-08 // Smallest positive half
+
+#    define HALF_NRM_MIN 6.10351562e-05 // Smallest positive normalized half
+
+#    define HALF_MIN 6.10351562e-05f // Smallest positive normalized half
+
+#    define HALF_MAX 65504.0 // Largest positive half
+
+#    define HALF_EPSILON 0.00097656 // Smallest positive e for which
+// half (1.0 + e) != half (1.0)
+#endif
+
+#define HALF_MANT_DIG                                                                              \
+    11 // Number of digits in mantissa                                                            \
+        // (significand + hidden leading 1)
+
+//
+// floor( (HALF_MANT_DIG - 1) * log10(2) ) => 3.01... -> 3
+#define HALF_DIG                                                                                   \
+    3 // Number of base 10 digits that                                                           \
+        // can be represented without change
+
+// ceil(HALF_MANT_DIG * log10(2) + 1) => 4.31... -> 5
+#define HALF_DECIMAL_DIG                                                                           \
```

Comment:
Hey guys, if you click and drag near the line numbers, it will select a block of code to comment. If you then click the [+-] icon on the left of the formatting tool bar it creates a suggestion block. Then, you can make an edit in place, like this, and all Kimball has to do is hit "accept suggestion". It would make it a lot easier for us to collaborate on this file if we start using the feature.

```suggestion
// Number of base 10 digits that can be represented without change
#define HALF_DIG 3

// ceil(HALF_MANT_DIG * log10(2) + 1) => 4.31... -> 5
#define HALF_DECIMAL_DIG                                                                           \
```